bndtools.core/src/bndtools/central/Central.java

```diff
@@ -204,39 +203,52 @@ public static Workspace getWorkspace() throws Exception {
             if (ws != null) { // early check for workspace
                 return ws;
             }
-            try {
-                Workspace.setDriver(Constants.BNDDRIVER_ECLIPSE);
-                Workspace.addGestalt(Constants.GESTALT_INTERACTIVE, new Attrs());
+            ws = initializeWorkspace();
+            // The workspace has been initialized fully, set the field now
+            workspace = ws;
+        }
+        workspaceQueue.resolve(ws); // notify onWorkspaceInit callbacks
```

Comment:
workspaceQueue is resolved with the original Workspace. If you reset the workspace field to a new Workspace, you still end up with all callers to onWorkspaceInit being run against the original workspace. If you change workspace fields, you also need to change the workspaceQueue field to a new Deferred which is resolved with the new workspace. This needs to be done properly. The current code is built around these fields being written once and then never changed. If you plan to change them, then a much different strategy is needed to handle the invariant that they being changed together. The onWorkspaceInit static method also needs to be changes to handle that the workspaceQueue field can change over time.